### ISSUE TITLE
pdns-recursor from GIT : compilation error on opensslsigners

### ISSUE THREAD

**Issue by Pascal76** [PowerDNS/pdns#4572]
on fedora 14/17 with latest gcc:

```
  CXX    gettime.o
  CXX    gss_context.o
  CXX    iputils.o
  CXX    ixfr.o
  CXX    json.o
  CXX    lua-recursor4.o
  CXX    lwres.o
  CXX    mtasker_context.o
  CXX    nsecrecords.o
  CXX    opensslsigners.o
opensslsigners.cc:131:41: erreur : 'ECDSA_SIG' does not name a type
 static inline void ECDSA_SIG_get0(const ECDSA_SIG* signature, const BIGNUM** pr, const BIGNUM** ps) {
                                         ^~~~~~~~~
opensslsigners.cc: Dans la fonction 'void ECDSA_SIG_get0(const int*, const BIGNUM**, const BIGNUM**)':
opensslsigners.cc:132:20: erreur : request for member 'r' in '* signature', which is of non-class type 'const int'
   *pr = signature->r;
                    ^
opensslsigners.cc:133:20: erreur : request for member 's' in '* signature', which is of non-class type 'const int'
   *ps = signature->s;
                    ^
opensslsigners.cc: At global scope:
opensslsigners.cc:136:34: erreur : 'ECDSA_SIG_set0' declared as an 'inline' variable
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                  ^~~~~~~~~
opensslsigners.cc:136:34: erreur : 'ECDSA_SIG' was not declared in this scope
opensslsigners.cc:136:45: erreur : 'signature' was not declared in this scope
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                             ^~~~~~~~~
opensslsigners.cc:136:62: erreur : expected primary-expression before '*' token
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                                              ^
opensslsigners.cc:136:64: erreur : 'pr' was not declared in this scope
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                                                ^~
opensslsigners.cc:136:74: erreur : expected primary-expression before '*' token
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                                                          ^
opensslsigners.cc:136:76: erreur : 'ps' was not declared in this scope
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                                                            ^~
opensslsigners.cc:136:78: erreur : expression list treated as compound expression in initializer [-fpermissive]
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                                                              ^
opensslsigners.cc:136:19: attention : 'ECDSA_SIG_set0' defined but not used [-Wunused-variable]
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                   ^~~~~~~~~~~~~~
make[2]: *** [opensslsigners.o] Erreur 1
make[2] : on quitte le répertoire « /root/pascal/pdns_recursor_git/pdns-master/pdns/recursordist »
make[1]: *** [all-recursive] Erreur 1
make[1] : on quitte le répertoire « /root/pascal/pdns_recursor_git/pdns-master/pdns/recursordist »
make: *** [all] Erreur 2
```

**Comment by Habbie** [comment]
1. I took the liberty of formatting your report
2. Can you please set your language to English when posting logs?
3. Why are you compiling on old (insecure) Fedora?
4. What version of OpenSSL?

**Comment by Pascal76** [comment]
2/ I will

3/ the company doesn't want to do the upgrade ...

4/
>openssl 
OpenSSL> version
OpenSSL 1.0.0e-fips 6 Sep 2011
OpenSSL> 

OpenSSL> version
OpenSSL 1.0.0k-fips 5 Feb 2013

**Comment by rgacogne** [comment]
According to https://bugzilla.redhat.com/show_bug.cgi?id=319901, ECDSA support was disabled until Fedora 19/20, where it was added in an OpenSSL update. While I don't think we care about supporting (very) old Fedora, we used to properly detect the lack of ECDSA support but it looks like we broke that at some point. Hopefully fixed by #4573.

**Comment by Pascal76** [comment]
it correct the problem
Thank you
